play-json/shared/src/main/scala/JsMacroImpl.scala

```diff
@@ -25,11 +25,39 @@ import scala.reflect.macros.blackbox
     }
   }
 
-  def readsImpl[A: c.WeakTypeTag, O <: Json.MacroOptions: c.WeakTypeTag]: c.Expr[Reads[A]] = macroImpl[A, Reads, Reads, O]("read", "map", reads = true, writes = false)
+  def withOptionsReadsImpl[A: c.WeakTypeTag]: c.Expr[Reads[A]] = macroImpl[A, Reads, Reads](
+    withOptionsConfig, "read", "map", reads = true, writes = false)
 
-  def writesImpl[A: c.WeakTypeTag, O <: Json.MacroOptions: c.WeakTypeTag]: c.Expr[OWrites[A]] = macroImpl[A, OWrites, Writes, O]("write", "contramap", reads = false, writes = true)
+  def withOptionsWritesImpl[A: c.WeakTypeTag]: c.Expr[OWrites[A]] = macroImpl[A, OWrites, Writes](
+    withOptionsConfig, "write", "contramap", reads = false, writes = true)
 
-  def formatImpl[A: c.WeakTypeTag, O <: Json.MacroOptions: c.WeakTypeTag]: c.Expr[OFormat[A]] = macroImpl[A, OFormat, Format, O]("format", "inmap", reads = true, writes = true)
+  def withOptionsFormatImpl[A: c.WeakTypeTag]: c.Expr[OFormat[A]] = macroImpl[A, OFormat, Format](
+    withOptionsConfig, "format", "inmap", reads = true, writes = true)
```

Comment:
These new variants are for invoking from `WithOptions.reads/writes/format`. They assume that the context they are invoked from as a `config: JsonConfiguration.Aux[Opts]` property on them.